src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs

```diff
@@ -683,5 +690,16 @@ private object ProcessCustomObject<T>(object o, int depth)
             }
             return result;
         }
+
+        /// <summary>
+        /// Check if Stopping is set and throw special exception to stop.
+        /// </summary>
+        private void CheckStopping()
+        {
+            if (Stopping)
+            {
+                throw new StopUpstreamCommandsException(this);
```

Comment:
I don't think we should use this exception. It's supposed to be thrown from a `ProcessRecord` method, to stop the upstream `Process` blocks from running. For the `ConvertTo-Json` cmdlet, we do the real work at `EndProcessing`, and by that time all upstream commands have finished running.

I suggest you use a custom exception instead, can catch the exception in `EndProcessing` when calling `ProcessValue`, like this:
```c#
object preprocessedObject = null;
try
{
    preprocessedObject = ProcessValue(objectToProcess, 0);
}
catch (MyCustomException ex)
{
    return;
}
```

Reply:
Makes sense, will update.